store/postgres/inter_project_job_spec_repository.go

```diff
@@ -0,0 +1,64 @@
+package postgres
+
+import (
+	"context"
+	"errors"
+
+	"gorm.io/gorm"
+
+	"github.com/odpf/optimus/models"
+	"github.com/odpf/optimus/store"
+)
+
+type interProjectJobSpecRepository struct {
+	db      *gorm.DB
+	adapter *JobSpecAdapter
+}
+
+func (repo interProjectJobSpecRepository) GetWithFilters(ctx context.Context, projectName, jobName, resourceDestination string) ([]models.JobSpec, error) {
+	var project Project
+	var jobs []Job
+	var specs []models.JobSpec
+	query := repo.db.WithContext(ctx).Preload("Namespace").Preload("Project")
+	if projectName != "" {
+		if err := repo.db.WithContext(ctx).Where("name = ?", projectName).First(&project).Error; err != nil {
+			if errors.Is(err, gorm.ErrRecordNotFound) {
+				return []models.JobSpec{}, store.ErrResourceNotFound
+			}
+			return []models.JobSpec{}, err
+		}
+		query = query.Where("project_id = ?", project.ID)
+	}
+	if jobName != "" {
+		query = query.Where("name = ?", jobName)
+	}
+	if resourceDestination != "" {
+		query = query.Where("destination = ?", resourceDestination)
+	}
+	if err := query.Find(&jobs).Error; err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
```

Comment:
We can remove this as in gorm v2, an empty result in the `Find` call will not return `ErrRecordNotFound`